### PR TITLE
Feature/refactor time utils

### LINKED ISSUE #288: Unify time formatting across whole app.
Right now we have formatting logic in `time.ts` and specific logic in `Table` and `LineChart` visualisation. We need to identify common parts and extract them. We also should consider if any inconsistencies are necessary and maybe we could simplify current behaviour.

### PR DESCRIPTION
**Pull request by adrianmroz**
Closes #288

### CODE REVIEW

**Comment by adrianmroz** [comment]
Also closes #274

File: src/common/utils/time/time.ts [hunk 1]
```diff
@@ -175,68 +148,41 @@ export function formatTimeElapsed(date: Date, timezone: Timezone): string {
...
 }
 
-export function formatGranularity(granularity: string): string {
-  return granularity.replace(/^PT?/, "");
-}
+const ISO_TIME_DISALLOWED = /[^\d:]/g;
 
-export function maybeFullyDefinedDate(date: string): boolean {
-  return date.length === FORMAT_DATE.length;
+export function trimISOTime(time: string): string {
```
Comment (piter75) [inline src/common/utils/time/time.ts]: I would change the verb to *prepare* or *normalize* as the name is a bit misleading now

**Comment by adrianmroz** [comment]
Also closes #279

File: src/common/utils/time/time.ts [hunk 2]
```diff
@@ -15,92 +15,124 @@
...
-const FORMAT_FULL_MONTH_WITH_YEAR = "MMMM YYYY";
+const FULL_FORMAT = "D MMM YYYY H:mm";
+const WITHOUT_YEAR_FORMAT = "D MMM H:mm";
+const WITHOUT_HOUR_FORMAT = "D MMM YYYY";
+const WITHOUT_YEAR_AND_HOUR_FORMAT = "D MMM";
+
+const SHORT_WITHOUT_HOUR_FORMAT = "D.MM.YY";
+const SHORT_FULL_FORMAT = "D.MM.YY HH:mm";
+const SHORT_WITHOUT_YEAR_FORMAT = "D.MM HH:mm";
+const SHORT_WITHOUT_YEAR_AND_HOUR_FORMAT = "D.MM";
```
Comment (l2dy) [inline src/common/utils/time/time.ts]: Why not put month before day following ISO convention?

**Pull request merged**